### DISCUSSION TITLE
kitty not working over ssh even using info from FAQ

### DISCUSSION THREAD

**Discussion by miguelcarcamov** [kovidgoyal/kitty#4874] [Q&A]
Hello everyone,

I have checked kitty's FAQ and there is a part where it says _I get errors about the terminal being unknown or opening the terminal failing when SSHing into a different computer?_

I have tried to ssh as mentioned in the FAQ but I get the following errors:

```
kitty +kitten ssh server
user@server's password: 
/bin/sh: Event not found.
#: Command not found.
#: Command not found.
Illegal variable name.
tmp: Undefined variable.
Connection to server closed.
```
```
kitty +kitten ssh use-python server
user@server's password: 
/usr/bin/env: Event not found.
from: Command not found.
from: Command not found.
import: unable to open X server `' @ error/import.c/ImportImageCommand/344.
#: Command not found.
#: Command not found.
Badly placed ()'s.
tname: Command not found.
if: Expression Syntax.
tname: Command not found.
Badly placed ()'s.
Badly placed ()'s.
Badly placed ()'s.
Badly placed ()'s.
if: Expression Syntax.
Badly placed (.
Badly placed ()'s.
Badly placed ()'s.
Badly placed ()'s.
except: Command not found.
shell_path: Command not found.
Badly placed ()'s.
if: Expression Syntax.
Badly placed ()'s.
Badly placed ()'s.
Unmatched ".
Connection to server closed.
```
```
infocmp -a xterm-kitty | ssh server tic -x -o \~/.terminfo /dev/stdin
user@server's password: 
tic: /dev/stdin is not a file
```

What can I do to make kitty work?

Cheers,

**Comment by kovidgoyal** [comment]
Copy the terminfo files manually. Do

kitty +kitten ssh localhost

that will create ~/.terminfo
Copy that directory to your remote server.

Though judging by all those errors, the lord alone knows what you are
running on that server so whether it will work, who knows.

**Reply by miguelcarcamov** [reply]
I have just done that. Getting same errors. It might be something that the server is doing though. Is there a way I can trace that?

**Comment by kovidgoyal** [comment]
ncurses reads terminfo from ~/.terminfo if you have put the
files there and you are still getting the error, then either 

1) you have not actually put the files there
2) the file permissions are incorrect
3) the error is coming from something other than ncurses
4) ncurses is broken on the server.

Use strace to debug.

**Comment by page-down** [comment]
> /bin/sh: Event not found.

I believe your user's login shell is tcsh or csh or whatever, not POSIX sh compatible.
Try chsh to change your user login shell.

> I have just done that. Getting same errors.

This means that you run `kitty +kitten ssh` again, which will not work.
What environment are you connected to? Linux? BSD?
You should send the terminfo db file to the remote host and compile it via tic. Or you can try the following.

```shell
ssh user@host "curl --create-dirs -o ~/.terminfo/x/xterm-kitty -fSL https://github.com/kovidgoyal/kitty/raw/v0.24.4/terminfo/x/xterm-kitty"
ssh user@host
```
